src/proto/network/snp.rs

```diff
@@ -0,0 +1,383 @@
+//! Network I/O protocols.
+
+use crate::proto::Protocol;
+use crate::{unsafe_guid, Result, Status};
+
+/// The Snp protocol.
```

Comment:
Could you mention the full name of the protocol in the doc comment? A brief description line would also be nice, if possible